src/riak_kv_get_fsm.erl

```diff
@@ -500,7 +485,9 @@ waiting_read_repair({r, VnodeResult, Idx, _ReqId},
         _ ->
             ok
     end,
-    UpdGetCore = riak_kv_get_core:add_result(Idx, VnodeResult, GetCore),
+    ResNode = find_node(Idx, StateData#state.preflist2),
```

Comment:
I hope that Idx is always present in preflight, even after a restart? Otherwise the add_node call on next line may fail.

Reply:
Yes, I'm happy this is true